analyzers/src/SonarAnalyzer.Common/Rules/BackslashShouldBeAvoidedInAspNetRoutesBase.cs

```diff
@@ -0,0 +1,75 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules;
+
+public abstract class BackslashShouldBeAvoidedInAspNetRoutesBase<TSyntaxKind> : SonarDiagnosticAnalyzer<TSyntaxKind>
+    where TSyntaxKind : struct
+{
+    private const string DiagnosticId = "S6930";
+
+    protected abstract TSyntaxKind[] SyntaxKinds { get; }
+
+    protected override string MessageFormat => @"Replace '\' with '/'.";
+
+    protected BackslashShouldBeAvoidedInAspNetRoutesBase() : base(DiagnosticId) { }
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterCompilationStartAction(compilationStartContext =>
+        {
+            if (ShouldRegisterAction(compilationStartContext.Compilation))
+            {
+                compilationStartContext.RegisterNodeAction(Language.GeneratedCodeRecognizer, Check, SyntaxKinds);
+            }
+        });
+
+    protected void Check(SonarSyntaxNodeReportingContext c)
+    {
+        if (Language.Syntax.NodeExpression(c.Node) is { } expression
+            && Language.FindStringConstant(c.SemanticModel, expression) is { } constantRouteTemplate
+            && ContainsBackslash(constantRouteTemplate)
+            && IsRouteTemplate(c.SemanticModel, c.Node, c.Node.Parent.Parent))
+        {
+            c.ReportIssue(Diagnostic.Create(Rule, expression.GetLocation()));
+        }
+    }
+
+    private bool IsRouteTemplate(SemanticModel model, SyntaxNode node, SyntaxNode invocation) =>
+        model.GetSymbolInfo(invocation).Symbol is IMethodSymbol methodSymbol
```

Comment:
Because of coverage: using `node.Parent.Parent is {} attributeInvocation` requires covering a case where `node.Parent.Parent` is null, and I couldn't find a test scenario where it is, with node being a `ArgumentSyntax` or an `AttributeArgumentSyntax`.

If the node is an `ArgumentSyntax` and the node is in the context of a method invocation, the `Parent` is always a `ArgumentListSyntax`, and the parent of the parent is always an `InvocationExpression`. For example, `f("\\d")` becomes:
![image](https://github.com/SonarSource/sonar-dotnet/assets/2676742/ca5c7ce5-05a2-4a66-bad7-8c8567540de8)

If the node is an `ArgumentSyntax` and the node is in the context of a tuple, the `Parent` is always a `TupleExpression`, and the parent of the parent is something like an `EqualsValueClause`. For example, `var x = (a: "\\d", b: "d");` becomes:
![image](https://github.com/SonarSource/sonar-dotnet/assets/2676742/726a51c2-2581-4779-b184-b87433e45736)

Even a tuple on a file with top-level statements is wrapped into a `TupleExpression`:
![Uploading image.png…]()

I had the impression that `node.Parent.Parent is var invocation` would trigger a "gratuitous condition", but I tried again now, and it seems to work. So, if you agree, I will change to that, since, in this specific scenario, it conforms with our code conventions:

> Var pattern is var o can be used only where variable declarations would require additional nesting.